jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/SetupAuthenticationDialog.kt

```diff
@@ -146,14 +153,14 @@ class SetupAuthenticationDialog(
     private val isFirstInstance: Boolean = false,
     private val connectionInitiatedFromExplorer: Boolean = false,
     private val connectionInitiatedFromQChatPanel: Boolean = false
-) : DialogWrapper(project) {
+) : DialogWrapper(project), AuthenticationDialog {
     private val rootTabPane = JBTabbedPane()
-    private val idcTab = idcTab()
-    private val builderIdTab = builderIdTab()
+    private val idcTab = IdcTabPanelBuilder(state.idcTabState::startUrl, state.idcTabState::region, profileName = Optional.of(state.idcTabState::profileName)).build()
+    private val builderIdTab = BuilderIdTabPanelBuilder().build()
     private val iamTab = iamTab()
     private val wrappers = SetupAuthenticationTabs.values().associateWith { BorderLayoutPanel() }
-    var attempts = 0
-    var authType = CredentialSourceId.IamIdentityCenter
+    private var attempts = 0
```

Comment:
prefer:
```suggestion
    var attempts = 0
        private set
```

Reply:
Does this suggestion apply to `authType` too?

Reply:
yes